src/librustdoc/core.rs

```diff
@@ -273,6 +274,58 @@ where
     (lint_opts, lint_caps)
 }
 
+crate fn get_rustdoc_lints(
+    cfg: FxHashSet<(String, Option<String>)>,
+    opts: config::Options,
+) -> Vec<String> {
+    let config = interface::Config {
+        opts,
+        crate_cfg: cfg,
+        input: Input::File(PathBuf::new()),
+        input_path: None,
+        output_file: None,
+        output_dir: None,
+        file_loader: None,
+        diagnostic_output: DiagnosticOutput::Default,
+        stderr: None,
+        lint_caps: Default::default(),
+        register_lints: None,
+        override_queries: None,
+        make_codegen_backend: None,
+        registry: rustc_driver::diagnostics_registry(),
+    };
+    let missing_docs_lints = rustc_lint::builtin::MISSING_DOCS.name;
+    let renamed_and_removed_lints = rustc_lint::builtin::RENAMED_AND_REMOVED_LINTS.name;
+    let unknown_lints = rustc_lint::builtin::UNKNOWN_LINTS.name;
+
+    // In addition to those specific lints, we also need to allow those given through
+    // command line, otherwise they'll get ignored and we don't want that.
```

Comment:
```suggestion
    // Rustdoc silences all lints by default, only warning on lints in the `rustdoc` group.
    // These lints aren't in the lint group, but we want to warn on them anyway.
```

Reply:
Also, if I understand correctly, rustdoc will no longer warn even if you add `-W non_upper_case_globals` or something like that, is that correct? I'm ok with that change personally, but it needs an FCP IMO.

Reply:
It currently doesn't. I tried with this code:

```rust
pub static foo: u32 = 0;

fn main() {}
```

Then I ran:

```bash
$ rustdoc foo.rs -W non_upper_case_globals
$ rustc foo.rs -W non_upper_case_globals
warning: static variable `foo` should have an upper case name
 --> foo.rs:1:12
  |
1 | pub static foo: u32 = 0;
  |            ^^^ help: convert the identifier to upper case (notice the capitalization): `FOO`
  |
  = note: requested on the command line with `-W non-upper-case-globals`

warning: 1 warning emitted
```

So no changes there as far as I can see.

Reply:
Just for info, here are the versions:

```bash
$ rustdoc --version
rustdoc 1.51.0-nightly (2987785df 2020-12-28)
$ rustc --version
rustc 1.51.0-nightly (2987785df 2020-12-28)
```